### PR TITLE
chore: handle `create fuels` template scenarios for insufficient funds

### LINKED ISSUE #2463: Handle `create fuels`  example dApp scenarios for insufficient funds
### Description

When a user submits a transaction with insufficient funds, they should be notified why the transaction hasn't been sent in an error modal with a link to redirect to the faucet

### PR DESCRIPTION
**Pull request by Dhaiwat10**
- Closes #2463

# Summary

This PR adds missing `try-catch` statements in some places where error handling is needed in case of insufficient funds in a wallet.

# Checklist

- [x] I **_added_** — `tests` to prove my changes
- [x] I **_updated_** — all the necessary `docs`
- [x] I **_reviewed_** — the entire PR myself, using the GitHub UI
- [x] I **_described_** — all breaking changes and the Migration Guide

### CODE REVIEW

File: apps/create-fuels-counter-guide/src/app/script/page.tsx [hunk 1]
```diff
@@ -39,7 +39,10 @@ export default function ScriptExample() {
       setResult(value.toString());
     } catch (error) {
       console.error(error);
-      toast.error("Error running script.");
+      toast.error(
+        `Error running script. Please make sure your wallet has enough funds.
```
Comment (maschad) [inline apps/create-fuels-counter-guide/src/app/script/page.tsx:43]: Perhaps we should hyperlink to the Top-up Wallet button here, is that possible with the `tost.error` ?

File: templates/nextjs/src/sway-api/contract-ids.json [hunk 2]
```diff
@@ -1,3 +1,3 @@
 {
-  "testContract": "dummy-contract-id"
-}
+  "testContract": "0x5cfe9ecc6b380e45e5b5f59e9badbd05c095c330c7d2bcb8da5a2f6bfe088c53"
```
Comment (maschad) [inline templates/nextjs/src/sway-api/contract-ids.json:2]: Shouldn't this be generated for each user?

**Review by maschad — COMMENTED** [review]
Looks good, comments inline

File: apps/create-fuels-counter-guide/src/app/predicate/page.tsx [hunk 3]
```diff
@@ -41,21 +41,29 @@ export default function PredicateExample() {
...
-    await refreshBalances();
+      await refreshBalances();
 
-    return toast.success("Funds transferred to predicate.");
+      return toast.success("Funds transferred to predicate.");
+    } catch (e) {
+      console.error(e);
+      toast.error(
+        `Failed to transfer funds. Please make sure your wallet has enough funds.
+        Please click the 'Top-up Wallet' button in the top right corner, or use the faucet in the top left.`,
```
Comment (danielbate) [inline apps/create-fuels-counter-guide/src/app/predicate/page.tsx:64]: Should we be checking for the specific error? As the toast error _could_ be unrelated to the console error.

Comment (Dhaiwat10) [reply apps/create-fuels-counter-guide/src/app/predicate/page.tsx]: @danielbate we _could_ be, but then there could be a lot of errors that we could be checking for. I don't think it's possible to check for this specific error yet until #2522 is worked upon

File: templates/nextjs/src/sway-api/contract-ids.json (hunk 2 above)
Comment (Dhaiwat10) [reply templates/nextjs/src/sway-api/contract-ids.json]: oops https://github.com/FuelLabs/fuels-ts/pull/2798/commits/c5612888786f357e1c1890b5a969f8c2bb8bd33c

File: apps/create-fuels-counter-guide/src/app/predicate/page.tsx (hunk 3 above)
Comment (Dhaiwat10) [reply apps/create-fuels-counter-guide/src/app/predicate/page.tsx]: Well we could just try and scrape the string 'not enough coins to fit the target' from the error message but that seems to be more of something that should be worked upon as a part of #2522. Anything we do to go around it here in the template will be a band-aid/hotfix

File: apps/create-fuels-counter-guide/src/app/script/page.tsx (hunk 1 above)
Comment (petertonysmith94) [reply apps/create-fuels-counter-guide/src/app/script/page.tsx]: Is it worth having a faucet link that will redirect to either the `local` or `testnet` based on env?

Comment (Dhaiwat10) [reply apps/create-fuels-counter-guide/src/app/script/page.tsx]: https://github.com/FuelLabs/fuels-ts/pull/2798/commits/e9ca5aefd2f468ff0ed7680ca9061a9bb0761fe8

File: apps/create-fuels-counter-guide/src/app/page.tsx [hunk 4]
```diff
@@ -49,7 +49,11 @@ export default function Home() {
 
     if (walletBalance?.eq(0)) {
       return toast.error(
-        "Your wallet does not have enough funds. Please click the 'Top-up Wallet' button in the top right corner, or use the local faucet.",
+        <span>
+          Your wallet does not have enough funds. Please click the 'Top-up
```
Comment (maschad) [inline apps/create-fuels-counter-guide/src/app/page.tsx:53]: I think we can just include the link.

```suggestion
Your wallet does not have enough funds. Please top it up with the {" "}
          <Link href={FAUCET_LINK} target='_blank'>faucet.</Link>
```

File: apps/create-fuels-counter-guide/src/app/predicate/page.tsx [hunk 5]
```diff
@@ -41,21 +42,32 @@ export default function PredicateExample() {
...
-    await refreshBalances();
+      await refreshBalances();
 
-    return toast.success("Funds transferred to predicate.");
+      return toast.success("Funds transferred to predicate.");
+    } catch (e) {
+      console.error(e);
+      toast.error(
+        <span>
+          Failed to transfer funds. Please make sure your wallet has enough
```
Comment (maschad) [inline apps/create-fuels-counter-guide/src/app/predicate/page.tsx:65]: Same here

File: apps/create-fuels-counter-guide/src/app/script/page.tsx [hunk 6]
```diff
@@ -39,7 +39,13 @@ export default function ScriptExample() {
       setResult(value.toString());
     } catch (error) {
       console.error(error);
-      toast.error("Error running script.");
+      toast.error(
+        <span>
+          Error running script. Please make sure your wallet has enough funds.
+          Please click the 'Top-up Wallet' button in the top right corner, or
```
Comment (maschad) [inline apps/create-fuels-counter-guide/src/app/script/page.tsx:45]: Same here

File: templates/nextjs/src/app/page.tsx [hunk 7]
```diff
@@ -47,7 +52,13 @@ export default function Home() {
 
     if (walletBalance?.eq(0)) {
       return toast.error(
-        "Your wallet does not have enough funds. Please click the 'Top-up Wallet' button in the top right corner, or use the local faucet.",
+        <span>
+          Your wallet does not have enough funds. Please click the 'Top-up
+          Wallet' button in the top right corner, or use the{" "}
+          <Link href={FAUCET_LINK} target="_blank">
```
Comment (maschad) [inline templates/nextjs/src/app/page.tsx:58]: Same here

File: templates/nextjs/src/app/predicate/page.tsx [hunk 8]
```diff
@@ -41,21 +42,35 @@ export default function PredicateExample() {
...
+      await refreshBalances();
 
-    return toast.success("Funds transferred to predicate.");
+      return toast.success("Funds transferred to predicate.");
+    } catch (e) {
+      console.error(e);
+      toast.error(
+        <span>
+          Failed to transfer funds. Please make sure your wallet has enough
+          funds. Please click the 'Top-up Wallet' button in the top right
```
Comment (maschad) [inline templates/nextjs/src/app/predicate/page.tsx:66]: Same here

File: apps/create-fuels-counter-guide/src/app/page.tsx (hunk 4 above)
Comment (Dhaiwat10) [reply apps/create-fuels-counter-guide/src/app/page.tsx]: https://github.com/FuelLabs/fuels-ts/pull/2798/commits/97b13d9951e7d282380ac61de620dc115b1b0ce2

File: apps/create-fuels-counter-guide/src/app/predicate/page.tsx (hunk 3 above)
Comment (Torres-ssf) [reply apps/create-fuels-counter-guide/src/app/predicate/page.tsx]: I agree with @danielbate here. We could try to verify if the error message is `not enough coins to fit the target`. 

But yeah, this means we would need to update the validation after https://github.com/FuelLabs/fuels-ts/issues/2522 is solved.

**Review by maschad — APPROVED** [review]
Nice work

**Review by arboleya — APPROVED** [review]

**Review by Torres-ssf — APPROVED** [review]

**Review by petertonysmith94 — APPROVED** [review]

**Pull request merged**